voltixApp/VoltixApp/Views/Send/EthereumTransactionListView.swift

```diff
@@ -1,173 +1,181 @@
-                        }
-                    }
-                } else if let errorMessage = etherScanService.errorMessage {
-                    Text("Error fetching transactions: \(errorMessage)")
-                } else {
-                    ProgressView()
-                }
-            }
-            .navigationTitle("Transactions")
-            .navigationBarBackButtonHidden()
-            .modifier(InlineNavigationBarTitleModifier())
-            .toolbar {
-                ToolbarItem(placement: .navigationBarLeading) {
-                    NavigationButtons.backButton(presentationStack: $presentationStack)
-                }
-                ToolbarItem(placement: .navigationBarTrailing) {
-                    NavigationButtons.questionMarkButton
-                }
-            }
-            .task {
-                if let vault = appState.currentVault {
-                    
-                    let result = EthereumHelper.getEthereum(hexPubKey: vault.pubKeyECDSA, hexChainCode: vault.hexChainCode)
-                    
-                    switch result {
-                        case .success(let eth):
-                            if let contract = contractAddress {
-                                await etherScanService.fetchERC20Transactions(
-                                    forAddress: eth.address,
-                                    contractAddress: contract
-                                )
-                                
-                            } else {
-                                await etherScanService.fetchTransactions(forAddress: eth.address)
-                            }
-                        case .failure(let error):
-                            print("error: \(error)")
-                    }
-                    
-                }
-            }
-        }
-    }
+	@StateObject var etherScanService: EtherScanService = .shared
+	@EnvironmentObject var appState: ApplicationState
+	@Binding var presentationStack: [CurrentScreen]
+	@State var contractAddress: String?
+	@State var addressFor: String = ""
+	@State var transactions: [EtherscanAPITransactionDetail] = []
+	
+	var body: some View {
+		VStack {
+			List {
+				if !transactions.isEmpty, !addressFor.isEmpty {
+					ForEach(transactions, id: \.hash) { transaction in
+						EthTransactionRow(transaction: transaction, myAddress: addressFor)
+					}
+					
+				} else {
+					ProgressView()
+				}
+			}
+			.navigationTitle("Transactions")
+			.navigationBarBackButtonHidden()
+			.modifier(InlineNavigationBarTitleModifier())
+			.toolbar {
+				ToolbarItem(placement: .navigationBarLeading) {
+					NavigationButtons.backButton(presentationStack: $presentationStack)
+				}
+				ToolbarItem(placement: .navigationBarTrailing) {
+					NavigationButtons.questionMarkButton
+				}
+			}
+			.task {
+				if let vault = appState.currentVault {
+					let result = EthereumHelper.getEthereum(hexPubKey: vault.pubKeyECDSA, hexChainCode: vault.hexChainCode)
```

Comment:
You should be able to grab the ETH coin from vault's Coins field

Reply:
I think we removed the link to this page, maybe we will need to add it later then we can pass the coin directly. Is that ok?